pdc_client/__init__.py

```diff
@@ -91,27 +91,71 @@ def read_config_file(server_alias):
 
 
 class PDCClient(object):
-    """BeanBag wrapper specialized for PDC access.
+    """
+    BeanBag wrapper specialized for PDC access.
 
     This class wraps general BeanBag.v1 objects, but provides easy-to-use
     interface that can use configuration files for specifying server
     connections. The authentication token is automatically retrieved (if
     needed).
     """
+
+    # Set a page_size which is used in _WrapperBeanBag
+    page_size = None
+
+    class _WrapperBeanBag(beanbag.BeanBag):
+        """
+        BeanBag wrapper for getting the constructor's page_size in used.
+
+        Re-implement the call function as in beanbag you can't call
+        super() function of parent class.
+        """
+
+        def call(self, path, *args, **kwargs):
```

Comment:
Don't copy-paste the method, use `super().call(...)` and docstring should just explain why is it overridden.

Reply:
This can't work, you can have a try in local env.

Reply:
And when you try use a super().func, will raise an error like the parent class is something like _WrapperBeanBagBase but not BeanBag, so super not works here. I saw some changes in deep of the BeanBag's code.

Reply:
OK, I think I understand. You use the derived class only as `self.client` but there is no documentation and certainty about type returned in `__getitem__()` and `__getattr__()`.

If that's the case then overriding methods of `BeanBag` is not way to go.

I think you have to wrap return values of `__getitem__()` and `__getattr__()` so you can override (as I mentioned [here](https://github.com/product-definition-center/pdc-client/pull/119#issuecomment-328505577)).